ur_description/urdf/ur_macro.xacro

```diff
@@ -1,25 +1,87 @@
+    the targetted robot.
+
+    If using the UniversalRobots/Universal_Robots_ROS_Driver, follow the steps
+    described in the readme to extract the kinematic calibration from the
+    controller and generate the required .yaml file.
+
+    Main author of the migration to yaml configs: Ludovic Delval.
+
+    Contributors to previous versions (in no particular order):
+
+     - Felix Messmer
+     - Kelsey Hawkins
+     - Wim Meeussen
+     - Shaun Edwards
+     - Nadia Hammoudeh Garcia
+     - Dave Hershberger
+     - G. vd. Hoorn
+     - Philip Long
+     - Dave Coleman
+     - Miguel Prada
+     - Marcel Schnirring
+     - Felix von Drigalski
+     - Felix Mauch
+     - Jimmy Da Silva
+     - Ajit Krisshna N L
+     - Muhammad Asif Rana
   -->
 
- <xacro:include filename="$(find ur_description)/urdf/ur.transmission.xacro" />
- <xacro:include filename="$(find ur_description)/urdf/ur.common.xacro" />
+  <xacro:include filename="$(find ur_description)/urdf/inc/ur_transmissions.xacro" />
+  <xacro:include filename="$(find ur_description)/urdf/inc/ur_common.xacro" />
+
+  <xacro:macro name="ur_robot" params="
+    prefix
+    joints_limits_parameters_file
+    kinematics_parameters_file
+    physical_parameters_file
+    visual_parameters_file
+    transmission_hw_interface:=hardware_interface/PositionJointInterface
+    safety_limits:=false
+    safety_pos_margin:=0.15
+    safety_k_position:=20"
+  >
+    <!-- Load configuration data from the provided .yaml files -->
+    <xacro:read_model_data
+      joints_limits_parameters_file="${joints_limits_parameters_file}" 
+      kinematics_parameters_file="${kinematics_parameters_file}"
+      physical_parameters_file="${physical_parameters_file}"
+      visual_parameters_file="${visual_parameters_file}"/>
 
- <xacro:macro name="ur_robot" params="prefix joints_limits_parameters_file kinematics_parameters_file physical_parameters_file visual_parameters_file 
-                  transmission_hw_interface:=hardware_interface/PositionJointInterface
-                  safety_limits:=false safety_pos_margin:=0.15 safety_k_position:=20">
+    <!-- Add URDF transmission elements (for ros_control) -->
+    <xacro:ur_arm_transmission prefix="${prefix}" hw_interface="${transmission_hw_interface}" />
 
-    <xacro:read_model_data joints_limits_parameters_file="${joints_limits_parameters_file}" 
-                    kinematics_parameters_file="${kinematics_parameters_file}"
-                    physical_parameters_file="${physical_parameters_file}"
-                    visual_parameters_file="${visual_parameters_file}"/>
+    <!-- This is for compatibility reasons, as transformation between base and
+         base_link was always 180 degree in the ur_description.
 
-    <!-- This is for compatibility reasons, as transformation between base and base_link was always 180 degree in the ur_description -->
+         NOTE: this will be removed in a future revision of this file
+     -->
```

Comment:
This was for example discussed inside https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/issues/78

But I guess this should be addressed in a different PR?

Reply:
Yes, I did not want to do that in this PR.